carbon-design-system/carbon-addons-iot-react
# fix(hierarchy-list): reset page (as necessary) when pageSize changes

Closes #2918 

**Summary**

- Previously, if you paged through the results to a higher page, but then changed the pagination (or removed it) You could get stuck on the higher page and be unable to get back to the list items.

**Change List (commits, features, bugs, etc)**

- add useEffect to check current page and max page and reset to page 1 if necessary.

**Acceptance Test (how to verify the PR)**

- Go the HierarchyList with large number of items story
- change pageSize to small
- change to page 2
- change pageSize to undefined
- you should return to page 1 with all results

**Regression Test (how to make sure this PR doesn't break old functionality)**

- no adverse effects on other stories



**Things to look for during review**
